saleor/payment/utils.py

```diff
@@ -427,3 +430,49 @@ def price_to_minor_unit(value: Decimal, currency: str):
     number_places = Decimal("10.0") ** precision
     value_without_comma = value * number_places
     return str(value_without_comma.quantize(Decimal("1")))
+
+
+def get_unfinished_payments():
+    """Return payments older than specified time and without assigned order object."""
+    ttl = settings.UNFINISHED_PAYMENT_TTL
+    now = timezone.now()
+    day_before = now - ttl
+
+    newest = Transaction.objects.filter(payment=OuterRef("pk")).order_by("-created")
+    payments = (
+        Payment.objects.get_queryset()
+        .filter(is_active=True, order=None)
+        .annotate(newest_trx_date=Subquery(newest.values("created")[:1]))
+        .filter(newest_trx_date__lte=day_before)
+    )
+
+    return payments
+
+
+def is_payment_unfinished_and_ready_to_release(payment: Payment):
+    """Payments older than specified time and without assigned order object."""
+    return get_unfinished_payments().filter(pk=payment.pk).exists()
+
+
+class ReleasePaymentException(Exception):
+    """Exception occured on attempt to release payment."""
+
+    pass
+
+
+def release_payment(payment: Payment):
+    """Try release payment calling refund or void."""
+    if not is_payment_unfinished_and_ready_to_release(payment):
+        raise ReleasePaymentException(f"Payment '{payment}' is not ready to release")
+    from ..plugins.manager import get_plugins_manager
+
+    manager = get_plugins_manager()
+    from saleor.payment.gateway import payment_refund_or_void
+
+    if payment.checkout is None:
```

Comment:
The payment method should have a `checkout` word inside a name as we raise validation error for payments witch doesn't have a checkout.

Reply:
Right, changed.